desktop/extensions-sdk/architecture/index.md

```diff
@@ -0,0 +1,116 @@
+service, for example:
+- run long-running processes that must outlive the frontend
+- store data in a local database and serve them back with a REST API
+- store the extension state, like when a button starts a long-running process, so that if you navigate away
+  from the extension and come back, the frontend can pick up where it left off
+- accessing specific resources in the Docker Desktop VM, for example by mounting folders in the compose
+file
+
+> **Tip**
+>
+> The `docker extension init` generates a Go backend. But you can still use it as a starting point for
+> your own extension and use any other language like Node.js, Python, Java, .Net, or any other language and framework.
+{: .tip }
+
+Usually, the backend is made of one container that runs within the Docker Desktop VM. Internally, Docker Desktop creates
+a Docker Compose project, create the container from the `image` option of the `vm` section of the `metadata.json`, and
+attaches it to the composer project. See the [ui metadata section](./metadata.md#vm) for more details.
+
+In many situations, extension backend services can be defined by using the same image also used to package the extension.
+This image must then have a defined `CMD` to start the backend service, in addition to `COPY` lines for the extension packaging.
+To ease version management, you can reuse the same image for the extension packaging and for the backend service and
+push it on Docker Hub.
+
+In some cases, a `composefile` can be used instead of an `image`. This is useful when the backend container
+needs more specific options, such as mounting volumes or requesting [capabilities](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities)
+that can't be expressed just with a Docker image. The `composefile` can also be used to add multiple containers
+needed by the extension, like a database or a message broker.
+
+> **Note**
+>
+> In some cases, it is useful to also interact with the Docker engine from the backend.
+> See [how to use the Docker socket](../guides/use-docker-socket-from-backend.md) from the backend.
+
+To communicate with the backend, the Extension SDK provides [functions](../dev/api/backend.md#get) to make `GET`,
+`POST`, `PUT`, and `DELETE` requests from the frontend. Under the hood, the communication is done through a socket
+or named pipe, depending on the operating system. If the backend was listening to a port, it would be difficult to
+prevent collision with other applications running on the host or in a container already. Also, some users are
+running Docker Desktop in constrained environments where they can't open ports on their machines.
+
+![Backend and frontend communication](./images/frontend-backend.png)
+
+Finally, the backend can be build with any technology of your choice, as long as it can run in a container and
+listen to socket or named pipe.
```

Comment:
```suggestion
Finally, the backend can be build with any technology, as long as it can run in a container and listen on a socket.
```

Named pipes are used on the host side, but extension developers never have to use them directly